include/deal.II/grid/grid_tools.h

```diff
@@ -2006,15 +2006,22 @@ namespace GridTools
 
   /**
    * Generates a partitioning of the active cells making up the entire domain
-   * using the same partitioning scheme as in the p4est library. After calling
+   * using the same partitioning scheme as in the p4est library if the flag
+   * blocked is set to true (default behavior of this function). After calling
    * this function, the subdomain ids of all active cells will have values
    * between zero and @p n_partitions-1. You can access the subdomain id of a
    * cell by using <tt>cell-@>subdomain_id()</tt>.
+   *
+   * @note If the flag blocked is set to false, some assumptions of the p4est
+   *       library are relaxed. E.g. p4est places all children of a cell on the
+   *       same processor if they are all active. By relaxing this, we can
+   *       can create partitions owning a single cell (also for refined meshes).
    */
   template <int dim, int spacedim>
   void
   partition_triangulation_zorder(const unsigned int            n_partitions,
-                                 Triangulation<dim, spacedim> &triangulation);
+                                 Triangulation<dim, spacedim> &triangulation,
+                                 const bool                    blocked = true);
```

Comment:
would you mind renaming this to ``group_siblings``, ``keep_siblings_on_one_process``, or something similar?